grails-app/services/com/unifina/service/DataUnionJoinRequestService.groovy

```diff
@@ -47,6 +56,13 @@ class DataUnionJoinRequestService {
 			}
 		}
 		sendMessage(c, "join")
+		ThreadUtil.sleep(1000)
+		if (!isMemberActive(c.contractAddress, c.memberAddress)) {
```

Comment:
Would be better to check every `interval` millis until a maximum of `timeout` has been reached.

Suitable params could be `interval = 1000` and `timeout = 10 * 1000`